### PR TITLE
[aotinductor] Enable test_non_default_cuda_device on CI

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110510
* __->__ #110509

Summary: test_non_default_cuda_device needs to run on a multi-gpu CI instance

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Differential Revision: [D49937115](https://our.internmc.facebook.com/intern/diff/D49937115)

### CODE REVIEW

**Review by angelayi — APPROVED** [review]

**Review by khabinov — APPROVED** [review]
Thx!

File: test/inductor/test_aot_inductor.py [hunk 1]
```diff
@@ -670,10 +683,11 @@ def forward(self, x, y):
         ):
             self.check_model(Repro(), example_inputs)
 
-    @unittest.skipIf(
-        torch.cuda.device_count() < 2, "The test requires multiple devices"
-    )
+    @requires_multigpu()
     def test_non_default_cuda_device(self):
+        if self.device == "cpu":
+            raise unittest.SkipTest("requires CUDA")
```
Comment (chenyang78) [inline test/inductor/test_aot_inductor.py:689]: Wondering if ```@requires_cuda()``` decorator would work or not?

**Review by chenyang78 — APPROVED** [review]

File: test/inductor/test_aot_inductor.py (hunk 1 above)
Comment (desertfire) [reply test/inductor/test_aot_inductor.py:689]: That will be a problem when running on a gpu instance, where both cuda and cpu version of this test will be run.

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49937115).

**Pull request closed without merging**